src/app/chat/views/channel.nim

```diff
@@ -241,8 +278,25 @@ QtObject:
     self.status.chat.unmuteChat(selectedChannel)
     self.updateChannelInRightList(selectedChannel)
 
+  proc unmuteChatItem*(self: ChannelView, id: string) {.slot.} =
+    if (self.chats.chats.len == 0): return
+    let selectedChannel = self.getChannelById(id)
+    if (selectedChannel == nil): return
+    if (selectedChannel.id == self.activeChannel.id):
+      self.unmuteCurrentChannel()
+      return
+    selectedChannel.muted = false
+    self.status.chat.unmuteChat(selectedChannel)
+    self.updateChannelInRightList(selectedChannel)
+
   proc channelIsMuted*(self: ChannelView, channelIndex: int): bool {.slot.} =
     if (self.chats.chats.len == 0): return false
     let selectedChannel = self.getChannel(channelIndex)
     if (selectedChannel == nil): return false
     result = selectedChannel.muted  
+
+  proc getChatItemById*(self: ChannelView, id: string): QObject {.slot.} =
+    let chat = self.getChannelById(id)
+    let chatItemView = newChatItemView(self.status)
+    chatItemView.setChatItem(chat)
+    return chatItemView
```

Comment:
This function is needed to get hold of a complete `QObject` chat item inside QML by a given id. Also paves the way to unify APIs and get rid off index based calls to this model

Reply:
I just realised, this is missing the memory table to ensure the returned object doesn't get out scope (and garbage collected). Will update